python/foglamp/services/core/scheduler/scheduler.py

```diff
@@ -940,10 +941,38 @@ def _schedule_row_to_schedule(cls,
         if not self._ready:
             raise NotReadyError()
 
+        class PseudoSchedule(object):
+            schedule_id = None
+            name = None
+            exclusive = True
+            enabled = False
+            repeat = None
+            time = None
+            day = None
+            process_name = None
+            schedule_type = None
+            group = None
+
+        cfg_manager = ConfigurationManager(self._storage_async)
+        get_children = await cfg_manager.get_all_children_with_parent()
+
         schedules = []
 
         for (schedule_id, schedule_row) in self._schedules.items():
-            schedules.append(self._schedule_row_to_schedule(schedule_id, schedule_row))
+            sch = self._schedule_row_to_schedule(schedule_id, schedule_row)
+            psch = PseudoSchedule()
+            psch.schedule_id = sch.schedule_id
+            psch.name = sch.name
+            psch.exclusive = sch.exclusive
+            psch.enabled = sch.enabled
+            psch.repeat = sch.repeat
+            psch.time = sch.time
+            psch.day = sch.day
+            psch.process_name = sch.process_name
+            psch.schedule_type = sch.schedule_type
+            # TODO: After scheduler bug is fixed, change process_name to name
+            psch.group = get_children[sch.process_name] if sch.process_name in get_children else ""
```

Comment:
This is mixing the concept of configuration hierarchy and schedules in a way that was never designed. The assumption that each schedule corresponds to a configuration category is probably not a good one.